department-of-veterans-affairs/abd-vro
# Mcp 1498

Getting all Lighthouse blood pressure and medication request data by paging

## Description
Modified code to accumulate all lighthouse pages.

### What was the problem?
When Lighthouse API has a large set of data, the default page size may not fit all the data it can return. So we need to use pagination to retrieve all the data.

### How does this fix it?
Call lighthouse FHIR API page by page and ccumulates all pages.

### Jira Tickets
MCP-1498

- [MCP-000](https://amida.atlassian.net/browse/MCP-000)

## How to test this PR

Step 1. Build and launch the application as stated in Readme file.

Step 2. Test the endpoint (http://localhost:8080/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#/Health%20Assessment/postHealthAssessment), in browser. Use the following test data.
diagnosticCode: "7101"
claimSubmissionId: use default.
veteranIcn: use any one in the list, [9000682, 32000225, 42000256, 18000027]

Step 3. Evaluate the responses. Should see something like this,

{
  "veteranIcn": "9000682",
  "diagnosticCode": "7101",
  "evidence": {
    "medications": [
      {
        "status": "Active",
        "notes": null,
        "description": "Hydrochlorothiazide 6.25 MG",
        "refills": 0,
        "duration": null,
        "authoredOn": "1950-04-06T04:00:00Z",
        "dosageInstructions": [
          null,
          "QD"
        ],
        "route": "ORAL"
      },
      {
        "status": "Active",
        "notes": [
          "Hydrochlorothiazide 6.25 MG"
        ],
        "description": "Hydrochlorothiazide 6.25 MG",
        "refills": 0,
        "duration": null,
        "authoredOn": "1950-04-06T12:24:55Z",
        "dosageInstructions": [
          "Once per day.",
          "As directed by physician."
        ],
        "route": "As directed by physician."
      }, ......

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
